src/persist-client/src/operators/shard_source.rs

```diff
@@ -253,11 +255,18 @@ where
     // just get the value directly with a function call.
     let mut pinned_stream = Box::pin(async_stream);
 
+    let empty_stream = timely::dataflow::operators::generic::operator::empty(scope);
+    let (flow_control_stream, flow_control_bytes) = match flow_control {
+        Some(fc) => (fc.progress_stream, Some(fc.max_inflight_bytes)),
```

Comment:
is there a downside to creating an operator and then not connecting it to the dataflow? if not, let's toss in a comment assuring a future code reader that this pattern is okay

Reply:
alternatively, Stream implements Clone, which would let us make flow_control_stream an owned `Stream` instead

Reply:
The downside is that it allocates an ID for the operator and stream, and then immediately shuts it down. It'd be more elegant to only construct `empty_stream` if it's needed and not unconditionally.

Reply:
I changed the `FlowControl` type to contain an owned `Stream` rather than a reference (in the spirit not hiding clones from the caller), and moved the creation of the `empty` operator into the match body.